packages/eslint-plugin/lib/rules/polaris-modal-requires-activator-prop.js

```diff
@@ -0,0 +1,34 @@
+const {docsUrl, polarisComponentFromJSX} = require('../utilities');
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Polaris <Modal /> and components that extend it must provide an `activator` prop to ensure proper keyboard focus handling on close',
+      category: 'Best Practices',
+      recommended: true,
+      uri: docsUrl('polaris-modal-requires-activator-prop'),
+    },
+    schema: [],
+  },
+
+  create(context) {
+    return {
+      JSXElement(node) {
+        const component = polarisComponentFromJSX(node, context);
+        if (
+          component === 'Modal' &&
+          node.openingElement.attributes.filter(
+            (attribute) => attribute.name.name === 'activator',
```

Comment:
Yeah, same here — it's kind of above my level a bit and I'm also still pretty new. Let me know if you'd like me to reach out cross-team or if it's something better handled by a staff / senior dev.

Reply:
I don't think anything is above our level here 😛 Just trying to think of the impact and potential implications of such a system-wide change before committing to it, that's why it's worth asking for feedback from teams that are directly involved with such changes while still advocating for _why_ we want to introduce this change and what other options we have considered to achieve this. Our end goal is to make the experience accessible and help others achieve the same with little to no effort as most of the tooling is already available to devs (focus managed if activator provided), sometimes we just need to bring awareness to it or enforce it.

Reply:
Thanks! I sometimes can be a bit overeager about reaching out across teams, so I mostly just wanted to make sure that you were suggesting it as an action for me rather than something you wanted to handle. I can explore the question a bit and get some feedback :)